src/netaddress.h

```diff
@@ -107,6 +109,22 @@ class CNetAddr
         }
 
         friend class CSubNet;
+        friend class CNetAddrHash;
+};
+
+class CNetAddrHash
+{
+public:
+    size_t operator()(const CNetAddr& a) const noexcept
+    {
+        CSipHasher hasher(salt_k0, salt_k1);
+        hasher.Write(a.ip, sizeof(a.ip));
+        return (size_t)hasher.Finalize();
```

Comment:
nit: All calls could be chained:
```suggestion
        return CSipHasher(salt_k0, salt_k1).Write(a.ip, sizeof(a.ip)).Finalize();
```

Reply:
I actually find the 3-line variant easier to read (that could be subjective). The 3-line variant is more manageable wrt setting a breakpoint or reading a backtrace (e.g. if you only see it crashed on that multi-chained-line it wouldn't be clear whether it crashed in the constructor, in the `Write()` method or in the `Finalize()` method).

I would prefer to leave it as is.